scenariolib/event_searchandclick_test.go

```diff
@@ -36,3 +32,116 @@ func TestSearchAndClickEventValid(t *testing.T) {
 	// Expect CustomData["data1"] to be "one"
 	equals(t, "one", event.CustomData["data1"])
 }
+
+func TestDecorateSearchAndClickEvent(t *testing.T) {
+	scenariolib.InitLogger(os.Stdout, os.Stdout, os.Stdout, os.Stderr)
+
+	expected := map[string]ExpectedRequest{
+		"/rest/search/": {
+			Method: "POST",
+			Headers: map[string]string{
+				"Authorization": "Bearer bot.searchToken",
+				"Content-Type":  "application/json",
+			},
+			Body: []byte(`{
+				"q": "aaaaaaaaaaa",
+				"numberOfResults": 20,
+				"tab": "All",
+				"pipeline": "besttechCommunity"
+			}`),
+		},
+		"/rest/v15/analytics/search/": {
+			Method: "POST",
+			Headers: map[string]string{
+				"Authorization": "Bearer bot.analyticsToken",
+				"Content-Type":  "application/json",
+			},
+			Body: []byte(`{
+					"language": "en",
+					"device": "Bot",
+					"customData": {
+						"JSUIVersion": "0.0.0.0;0.0.0.0",
+						"customData1": "customValue 1",
+						"ipaddress": "216.249.112.8"
+					},
+					"anonymous": true,
+					"originLevel1": "BotSearch",
+					"originLevel2": "",
+					"searchQueryUid": "",
+					"queryText": "aaaaaaaaaaa",
+					"actionCause": "searchboxSubmit",
+					"contextual": false
+				}`),
+		},
+	}
+
+	server := createMockServer(t, expected)
+	defer server.Close() // Close the server when test finishes
+
+	event := &scenariolib.SearchAndClickEvent{}
+	conf, err := scenariolib.NewConfigFromPath("../scenarios_examples/TESTScenarios.json")
+	ok(t, err)
+
+	// Use the server url to define the endpoints
+	conf.SearchEndpoint = server.URL + "/rest/search/"
+	conf.AnalyticsEndpoint = server.URL + "/rest/v15/analytics/"
```

Comment:
Can you make a constant for the version of analytics please?

Reply:
done